docs/DevGuide/BuildSystem.md

```diff
@@ -186,3 +186,38 @@ cmake -D FLAG1=OPT1 ... -D FLAGN=OPTN <SPECTRE_ROOT>
   - This needs to be turned `OFF` in order to use
     [include-what-you-use
     (IWYU)](https://github.com/include-what-you-use/include-what-you-use)
+
+## Formaline
+
+SpECTRE's implementation of Formaline is based on the original design by
+[Erik Schnetter and Christian Ott](https://github.com/hypercott/formaline),
+which embeds an archive of the source tree into the executable. The original
+design creates a C/C++ file with a function that returns an array/vector of
+`char`s (a byte stream). However, this results in a very large source file (50MB
+or more), which is very slow to compile and ends up more than doubling the link
+time. The linker `ld` supports encoding a file into an object, which means
+rather than creating a large source file, we can directly encode the source tree
+archive into the binary at the linking stage. Most of Formaline is implemented
+inside the `tools/WrapLinker.sh` script. Function declarations are provided
```

Comment:
Tangential thought: would it make improve modularity/comprehension to put the Formaline-related functions a `tools/Formaline.sh` file that can be used by `WrapLinker.sh`?

Reply:
I'm not sure :)

Here are the pros I can think of:
- code is factored better
- easier to read, maintain, and reuse

Cons:
- Then script needs to stay in sync with `WrapLinker.sh` since that uses the variables defined in the file `${formaline_output}`, `${formaline_dir}`, and `${archive_name}`. @wthrowe what would be the best way to get those variables out of the `Formaline.sh`? Maybe sourcing the `Formaline.sh` is all that's necessary, but my bash game is too weak :P

Reply:
Yes, this seems like a good plan.  For encapsulation sake, I would recommend passing `$(basename "${!oindex}")` as an argument to the script.  (You can pass arguments to sourced scripts in the same way as for a normal program.)  Also maybe rename `archive_dir` to begin with `formaline_` so it is clear that it is coming from the sourced script, or define a variable containing `${formaline_dir}/${archive_name}`, since it looks like that is all the external code needs.